main.ml

```diff
@@ -346,9 +346,13 @@ let split_file =
 let compile_script ?log source_option =
   let meta_name, prog_name = split_file ?log source_option in
   try Pipeline.run_command (Pipeline.command ["ocaml"; meta_name])
-  finally (* comment out for debugging: *)
-    Pipeline.remove meta_name;
-    Pipeline.remove prog_name
+  with exn ->
+    begin (* finally *)
+      (* comment out for debugging: *)
+      Pipeline.remove meta_name;
+      Pipeline.remove prog_name;
+      raise exn
+    end
```

Comment:
The "finally" block must be called also if no exception is raised. You should use `Fun.protect` for this. See https://caml.inria.fr/pub/docs/manual-ocaml/libref/Fun.html

Reply:
Ok, I will use that. But then it wil add a constraint (ocaml >= 4.08).

Reply:
Sounds reasonable. Otherwise you can define your own `protect` function.

Reply:
Something like this:
```ocaml
let protect ~finally f =
  try let res = f () in
  finally ();
  res
with e ->
  finally ();
  raise e
```

```
# protect ~finally:(fun () -> print_endline "bye") (fun () -> print_endline "hello"; raise Exit);;
hello
bye
Exception: Stdlib.Exit.
```